pandas/core/arrays/categorical.py

```diff
@@ -1081,20 +1081,26 @@ def remove_unused_categories(self, inplace=False):
             return cat
 
     def map(self, mapper):
-        """Apply mapper function to its categories (not codes).
+        """
+        Map categories (not codes) using input correspondence (a dict,
+        Series, or function).
+
+        Maps the categories to new categories. If the mapping
```

Comment:
I realize this is mostly a copy / paste of the original but I find the wording rather confusing - any thoughts on how to make the explanation clearer?

Reply:
I agree with you, this is not very clear but I don't know how to make it better. In my understanding a Categorical resembles in some respect a MultiIndex: the categories are a little bit like the levels and the codes are like the labels. Without that analogy in mind, any wording is going to be confusing.
What about something like that (a bit verbose):

"""
Maps the categories (the values of the elements of the array, not the
`codes` that represent the positions) to new categories. If the mapping
correspondence ...
"""

I added backticks to the `codes` to make it clear that we are talking about an attribute and not just a generic word.
What do you think? I'm open to suggestions!

Reply:
@WillAyd What is the aspect you find confusing? The first "Maps the categories to new categories" or the explanation after it?

Reply:
Do we have news on this?

Reply:
The first line still need to be consolidated to one line. 

As to the Extended Summary I think saying things like "mapping correspondence maps each original category to a different new category" is just very confusing. Could we not be more concise and just say something like "Maps the categories to new categories, retaining any ordering. Missing entries will be replaced with an `np.ndarray` containing `NaN`."?

I think that is the what the summary is trying to say now but it's not entirely clear. Adding Examples to show how the mapping works and how missing data is handled would make things a lot clearer.